definitions/external-service/definition.yml

```diff
@@ -0,0 +1,16 @@
+domain: EXTERNAL
+type: SERVICE
+
+synthesis:
+  rules:
+    # DistributedTraceProcessSummary events
+    - identifier: entity.guid
```

Comment:
This is something that I don't think is what you would expect. This represents the domain id. See [this doc](https://pages.datanerd.us/site-engineering/nr-platform-docs/entity-platform/entity-indexing/entity-indexing.html#entity-global-unique-identifier) for more information. It's not the guid that the synthesized entity would have, but just one of the 4 sections inside an entity guid. Same for the name.

You need to find which is the name and id that are coming along in the telemetry, and use those attributes as the source of the potential entity guid and entity name.

Reply:
entity.guid on DistributedTraceProcessSummary is the complete, encoded entity guid.
The domain ID upon which that entity guid is based depends on the entity type. When the entity is an EXTERNAL SERVICE, the entity.name is the host name, which is used for the domain ID in the entity guid.